motor/render/vulkan_renderer.cpp

```diff
@@ -355,78 +631,173 @@ class VulkanRenderer : public Renderer
   VulkanRenderer()
```

Comment:
additions to this function seems to be a lot, could you try separating them as well?